include/exadg/incompressible_navier_stokes/spatial_discretization/spatial_operator_base.cpp

```diff
@@ -195,23 +195,19 @@ SpatialOperatorBase<dim, Number>::setup(
   // initialize data container for DirichletCached boundary conditions
   if(not(boundary_descriptor->velocity->dirichlet_cached_bc.empty()))
   {
-    interface_data_dirichlet_cached = std::make_shared<ContainerInterfaceData<1, dim, double>>();
     std::vector<unsigned int> quad_indices;
     quad_indices.emplace_back(get_quad_index_velocity_linear());
     quad_indices.emplace_back(get_quad_index_velocity_nonlinear());
     quad_indices.emplace_back(get_quad_index_velocity_gauss_lobatto());
 
-    std::set<dealii::types::boundary_id> bids;
-    for(auto const & bc : boundary_descriptor->velocity->dirichlet_cached_bc)
-      bids.insert(bc.first);
-
-    interface_data_dirichlet_cached->setup(matrix_free,
-                                           get_dof_index_velocity(),
-                                           quad_indices,
-                                           bids);
+    interface_data_dirichlet_cached = std::make_shared<ContainerInterfaceData<1, dim, double>>();
+    interface_data_dirichlet_cached->template setup<Number>(
```

Comment:
interesting syntax here - looked it up and learned something, thanks ; )

Reply:
see e6393c33de4851c38486cf319fbb604a10e66d39